src/Driver/PostGISPlatform.php

```diff
@@ -0,0 +1,142 @@
+        $spatialIndexes = SpatialIndexes::extractSpatialIndicies($table->getIndexes());
+        foreach ($spatialIndexes as $index) {
+            $table->dropIndex($index->getName());
+        }
+
+        $sql = parent::getCreateTableSQL($table, $createFlags);
+
+        $spatialIndexSqlGenerator = new SpatialIndexSqlGenerator($this);
+        foreach ($spatialIndexes as $index) {
+            $sql[] = $spatialIndexSqlGenerator->getSql($index, $table);
+        }
+
+        return $sql;
+    }
+
+    public function getCreateTablesSQL(array $tables): array
+    {
+        $sql = [];
+        $spatialIndexSqlGenerator = new SpatialIndexSqlGenerator($this);
+
+        /** @var Table $table */
+        foreach ($tables as $table) {
+            SpatialIndexes::ensureSpatialIndexFlags($table);
+
+            $spatialIndexes = SpatialIndexes::extractSpatialIndicies($table->getIndexes());
+            foreach ($spatialIndexes as $index) {
+                $table->dropIndex($index->getName());
+            }
+            $sql = [...$sql, ...$this->getCreateTableWithoutForeignKeysSQL($table)];
+
+            foreach ($spatialIndexes as $index) {
+                $table->addIndex($index->getColumns(), $index->getName(), $index->getFlags(), $index->getOptions());
+            }
+
+            foreach ($spatialIndexes as $spatialIndex) {
+                $sql[] = $spatialIndexSqlGenerator->getSql($spatialIndex, $table);
+            }
+        }
+
+        foreach ($tables as $table) {
+            foreach ($table->getForeignKeys() as $foreignKey) {
+                $sql[] = $this->getCreateForeignKeySQL(
+                    $foreignKey,
+                    $table->getQuotedName($this),
+                );
+            }
+        }
+
+        return $sql;
+    }
+
+    public function getAlterTableSQL(TableDiff $diff): array
+    {
+        $table = $diff->getOldTable();
+        $spatialIndexSqlGenerator = new SpatialIndexSqlGenerator($this);
+
+        SpatialIndexes::ensureSpatialIndexFlags($diff);
+
+        $sql = parent::getAlterTableSQL(SpatialIndexes::filterTableDiff($diff));
+
+        foreach (SpatialIndexes::extractSpatialIndicies($diff->getAddedIndexes()) as $spatialIndex) {
+            $sql[] = $spatialIndexSqlGenerator->getSql($spatialIndex, $table);
+        }
+
+        foreach (SpatialIndexes::extractSpatialIndicies($diff->getModifiedIndexes()) as $index) {
+            $sql[] = $this->getDropIndexSQL($index->getName(), $table->getName());
+            $sql[] = $spatialIndexSqlGenerator->getSql($index, $table);
+        }
+
+        /** @var ColumnDiff $columnDiff */
+        foreach ($diff->getModifiedColumns() as $columnDiff) {
+            if(!$columnDiff->getOldColumn()->hasPlatformOption('srid') && !$columnDiff->getNewColumn()->hasPlatformOption('srid')) {
+                continue;
+            }
+            if ($columnDiff->getOldColumn()->getPlatformOption('srid') !== $columnDiff->getNewColumn()->getPlatformOption('srid')) {
```

Comment:
This check doesn't seem to work in our case.

`$columnDiff->getNewColumn()` has the `srid` platform option, but `$columnDiff->getOldColumn()` doesn't have it.
This makes it fail with the error `Warning: Undefined array key "srid"`.

I don't know why `$columnDiff->getOldColumn()` doesn't have the `srid` platform option at this point, though.

Reply:
I've updated the logic, does that fix it for you?

Reply:
I've poked and prodded, but I can't reproduce. I hit something like this when I was first composing the PR … but now :worried: 

Do you have the time to generate a test case? I'll happily dig in from there!